esp32/mods/modpycom.c

```diff
@@ -178,17 +212,53 @@ STATIC mp_obj_t mod_pycom_nvs_set (mp_obj_t _key, mp_obj_t _value) {
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_2(mod_pycom_nvs_set_obj, mod_pycom_nvs_set);
 
+
+
+STATIC mp_obj_t mod_pycom_nvs_getstring (mp_obj_t _key) {
+    vstr_t vstr;
+    const char *key = mp_obj_str_get_str(_key);
+    size_t required_size = 0;
+    if (ESP_ERR_NVS_NOT_FOUND == nvs_get_str(pycom_nvs_handle, key, NULL, &required_size)) {
+        return mp_const_none;
+    }
+    vstr_init_len(&vstr, required_size);
+    if (ESP_OK != nvs_get_str(pycom_nvs_handle, key, vstr.buf, &required_size)) {
```

Comment:
Instead of this, after the necessary size is fetched I would simply allocate enough byte with m_malloc(), fetch the value from NVS, then create the return object with  returned string with mp_obj_new_str(), then free up the allocated memory and return the composed string. In this case it is not needed to compose vstr, remove closing \0 etc.